README
升级流程

 # 上传3RMT程序到指定目录,导入数据库结构,设置程序路径和数据库
 # 关闭会员注册&订单
 # 备份数据库,和程序
 # 导入备份数据库,设置导入脚本的源备份数据库和源备份程序
 # 执行导入脚本auto.sh
 # 执行升级脚本upgrade.sql
 # 设置权限
 # 快速测试
 # 切换域名
 # 快速测试

script/history
  137  ls -lhs
  138  ls -lhS
  139  rm -fr everything.log
  140  /etc/rc.d/nfs-common start
  141  vim everything.log
  142  ls -lhSt
  143  vim /etc/conf.d/nfs-common.conf
  144  vim /etc/rpc 
  145  wget https://mail.google.com/mail/?ui=2&ik=63d4797039&view=att&th=126266f0fc43d00c&attid=0.1&disp=inline&realattid=f_g4dqzurz0&zw
  146  mogrify 
  147  sudo pacman -Ss mogrify
  148  convert
  149  sudo pacman -S imagemagick
  150  mogirfy
  151  mogrify
  152  mogrify -format ico logo_ico.gif
  153  sftp bobhero@192.168.2.111
  154  cd gti
  155  cd git
  156  ssh root@192.168.1.149
  157  ssh root@192.168.2.149
  158  vim index.php 
  159  exit
  160  httpd
  161  vim includes/application_top.php
  162  l;s
  163  /etc/rc.d/httpd restart
  164  vim /etc/httpd/conf/extra/httpd-vhosts.conf
  165  vim /etc/httpd/conf/httpd.conf 
  166  vim includes/configure.php 
  167  vim includes/boxes/best_sellers.php 
  168  vim includes/application_top.php 
  169  vim admin/includes/application_top.php 
  170  echo "" red-error_log 
  171  chmod 777 red-error_log 
  172  /etc/rc.d/httpd restart
  173  more red-error_log 
  174  LS
  175  PWD
  176  cd /home/bobhero
  177  rm -fr redmine
  178  tar zxvf redmine.tar
  179  mv redmine /usr/lib/re
  180  cd /usr/lib
  181  mv redmine rdbk
  182  mv re redmine
  183  cd config
  184  vim database.yml 
  185  cd db
  186  find ./ -name '*.sql'
  187  cd lib
  188  mv redmine rejp
  189  ls |grep re
  190  ls |grep reb
  191  mv rdbk redmine
  192  chmod 777 -R redmine
  193  sudo vi /etc/httpd/conf/extra/httpd-vhosts.conf
  194  sudo vi /var/log/httpd/red-error_log 
  195  sudo vi /usr/lib/redmine/log/production.log 
  196  vim redmine.log 
  197  sudo vi /etc/httpd/conf/extra/httpd-redmine.conf 
  198  cd /var/log
  199  cd httpd/
  200  vim red-error_log 
  201  vim access_log
  202  cd redmine
  203  more redmine.log 
  204  cd /usr/lib/redmine/
  205  cd log
  206  more production.log 
  207  yaourt -R redmine-svn
  208  yaourt redmine-mysql-svn
  209  yaourt -R redmine-mysql-svn
  210  yaourt redmine
  211  :gem install fastthread
  212  /usr/bin/gem install fastthread
  213  yaourt redmine
  214  :gem install fastthread
  215  /usr/bin/gem install fastthread
  216  vim index.php 
  217  vim includes/application_top.php 
  218  yaourt redmine
  219  :gem install fastthread
  220  /usr/bin/gem install fastthread
  221  yaourt redmine
  222  :gem install fastthread
  223  /usr/bin/gem install fastthread
  224  vim stylesheet.css 
  225  vim index.php 
  226  vim includes/application_top.php 
  227  vim includes/header.php 
  228  vim design.css 
  229  lsl
  230  lsl
  231  vim index.php 
  232  /usr/bin/gem install fastthread
  233  vim index.php 
  234  vim includes/application_top.php 
  235  mv newwm_test newwm_test03
  236  rm -rf newwm_test
  237  cp -rf newwm   newwm_test
  238  cp newwm_test03/admin/includes/configure.php newwm_test/admin/includes/
  239  cp newwm_test03/includes/configure.php newwm_test/includes/
  240  vim newwm_test/includes/configure.php 
  241  vim newwm_test/admin/includes/configure.php 
  242  cp newwm/design.css  newwm_test/
  243  cp newwm/design.css  newwm_test/
  244  vim includes/application_top.php 
  245  vim index.php 
  246  vim includes/application_top.php
  247  cd project/
  248  vim script/__database.php 
  249  php __database.php
  250  php __database.php > log
  251  mv newwm_test newwm_test04
  252  cp -rf newwm newwm_test
  253  cp  newwm_test04/includes/configure.php newwm_test/includes/
  254  cp newwm_test04/admin/includes/configure.php newwm_test/admin/includes/
  255  cd ../
  256  cd patch/
  257  git status|more
  258  php __database.php 
  259  vim __database.php 
  260  exit
  261  mv newwm_test newwm_test03
  262  rm -rf newwm_test
  263  cp -rf newwm   newwm_test
  264  cp newwm_test03/admin/includes/configure.php newwm_test/admin/includes/
  265  cp newwm_test03/includes/configure.php newwm_test/includes/
  266  vim newwm_test/includes/configure.php 
  267  vim newwm_test/admin/includes/configure.php 
  268  cp newwm/design.css  newwm_test/
  269  cp newwm/design.css  newwm_test/
  270  vim index.php 
  271  vim includes/application_top.php
  272  cd project/
  273  vim script/__database.php 
  274  php __database.php
  275  php __database.php > log
  276  mv newwm_test newwm_test04
  277  cp -rf newwm newwm_test
  278  cp  newwm_test04/includes/configure.php newwm_test/includes/
  279  cp newwm_test04/admin/includes/configure.php newwm_test/admin/includes/
  280  cd ../
  281  cd patch/
  282  git status|more
  283  php __database.php 
  284  vim __database.php 
  285  vim product_reviews_info.php 
  286  vim shopping_cart.php 
  287  vim login.php 
  288  vim checkout_products.php 
  289  vim checkout_payment.php 
  290  vim checkout_confirmation.php 
  291  vim checkout_success.php 
  292  vim reorder2.php 
  293  vim reorder.php 
  294  vim advanced_search.php 
  295  vim account_history.php 
  296  vim account.php 
  297  vim account_edit.php 
  298  vim product_notifications.php 
  299  vim password_forgotten.php 
  300  vim reviews.php 
  301  vim latest_news.php 
  302  vim account_history_info.php 
  303  vim create_account.php 
  304  vim create_account_process.php 
  305  vim account_edit_process.php 
  306  vim checkout_shipping.php 
  307  vim product_reviews.php 
  308  vim tell_a_friend.php 
  309  vim product_reviews_write.php 
  310  vim contact_us.php 
  311  vim send_mail.php 
  312  vim email_trouble.php 
  313  vim browser_ie6x.php 
  314  find ./ -name "*.php"|xargs grep 'function page_head'
  315  vim includes/languages/japanese/advanced_search_result.php 
  316  vim advanced_search_result.php 
  317  vim includes/functions/general.php 
  318  vim specials.php 
  319  vim manufacturers.php 
  320  vim .htaccess
  321  vim includes/application_top.php 
  322  vim game_news.php 
  323  exit
  324  ln -s /home/kthiz/project/OSC_FS kthizpf.200.com
  325  exit

gm/faq.php
<?php
/*
  osCommerce, Open Source E-Commerce Solutions
  http://www.oscommerce.com

  Copyright (c) 2003 osCommerce

  Released under the GNU General Public License
*/
require('includes/application_top.php');

$gc_name = '';
$faq_meta_k = '';
$faq_meta_d = '';
$h1_mes = '';
$faq_mes = '';
$breadcrumb->add('よくある質問', tep_href_link('info-7.html'));

if(isset($_GET['faq_id'])) {
	$faq_no = explode("/", $faq_id, 2);
	$g_id = $faq_no[0];
	if($g_id == "168" || $g_id == "169" || $g_id == "170" || $g_id == "171" || $g_id == "177" || $g_id == "178" || $g_id == "179" || $g_id == "190" || $g_id == "195") {
		$q_no = $faq_no[1];
		$q_id = rtrim($q_no, ".html");
		
		$categories_query = tep_db_query("select categories_name from categories_description where categories_id = '".$g_id."' and language_id = '".$languages_id."'");
		$categories = tep_db_fetch_array($categories_query);
		$gc_name = $categories['categories_name'];
		
		require('iimy_faq/' . (int)$g_id . '/faq.php');
		
		$breadcrumb->add($gc_name . 'のよくある質問', tep_href_link('faq' . (int)$g_id . '/'));
		$h1_mes = $gc_name . 'のよくある質問（FAQ)';
	} else {
		$faq_title = 'ファイルが見つかりません';
		$breadcrumb->add('エラー');
		$h1_mes = 'エラー';
		$faq_mes = '<h3>ファイルが見つかりません</h3><p>削除されたかURLが間違っております。</p>';
	}
}
?>
<?php page_head();?>
</head>
<body>
	<!-- header //--> 
	<?php require(DIR_WS_INCLUDES . 'header.php'); ?> 
	<!-- header_eof //--> 
	<!-- body //--> 
	<div id="main">
		<!-- left_navigation //-->
		<div id="l_menu">
			<?php require(DIR_WS_INCLUDES . 'column_left.php'); ?>
		</div>
		<!-- left_navigation_eof //-->
		<!-- body_text //-->
		<div id="content">
			<div class="box">
				<div class="headerNavigation"><?php echo $breadcrumb->trail(' &raquo; '); ?></div>
				<h1 class="pageHeading"><?php echo $h1_mes; ?></h1>
				<?php echo $faq_mes; ?>
			 </div>
		</div>
		<!-- body_text_eof //-->
		<div id="r_menu">
			<?php require(DIR_WS_INCLUDES . 'column_right.php'); ?> 
		</div> 
		<!-- body_eof //--> 
		<!-- footer //--> 
		<?php require(DIR_WS_INCLUDES . 'footer.php'); ?> 
		<!-- footer_eof //--> 
	</div>
</body>
</html>
<?php require(DIR_WS_INCLUDES . 'application_bottom.php'); ?>